File: pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.cpp
class Solution 
{
public:
    int numPairsDivisibleBy60(vector<int>& time) 
    {
        vector<int> m(60);
        int res = 0;
        for (auto t : time) 
        {
            res += m[(60 - t % 60) % 60];
            m[t % 60]++;
        }
        return res;
    }
};

File: pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        m = [0] * 60
        res = 0
        for t in time: 
            res += m[(60 - t % 60) % 60]
            m[t % 60] += 1
        return res